Detections/AuditLogs/GuestAccountsAddedinAADGroupsOtherThanTheOnesSpecified.yaml

```diff
@@ -0,0 +1,53 @@
+id: 1caa690e-4d5c-4083-ba1a-97cdebad4eff
+name: Guest accounts added in AAD Groups other than the ones specified
+description: |
+  'Guest Accounts are added in the Organization Tenants to perform various tasks i.e projects execution, support etc.. This detection notifies when guest users are added to Azure AD Groups other than the ones specified and poses a risk to gain access to sensitive apps or data.'
+severity: High
+requiredDataConnectors:
+  - connectorId: AzureActiveDirectory
+    dataTypes:
+      - AuditLogs
+queryFrequency: 5m
+queryPeriod: 1d
+triggerOperator: gt
+triggerThreshold: 0
+status: Available
+tactics:
+  - InitialAccess
+  - Persistence
+  - Discovery
+relevantTechniques:
+  - T1078.004
+  - T1136.003
+  - T1087.004
+query: |
+  let GroupIDs = dynamic(["List with Custom AAD GROUP OBJECT ID 1","Custom AAD GROUP OBJECT ID 2"]);
+  let queryfrequency = 5m;
+  let queryperiod = 1d;
+  AuditLogs
+  | where TimeGenerated > ago(queryperiod)
+  | where OperationName in ('Add member to group', 'Add owner to group')
+  | extend InitiatedByActionUserInformation = iff(isnotempty(InitiatedBy.user.userPrincipalName), InitiatedBy.user.userPrincipalName, InitiatedBy.app.displayName)
+  | extend InitiatedByIPAdress = InitiatedBy.user.ipAddress 
+  // Uncomment the following line to filter events where the inviting user was a guest user
+  //| where InitiatedBy has_any ("CUSTOM DOMAIN NAME#", "#EXT#")
+  | extend InvitedUser = TargetResources[0].userPrincipalName
+  | extend AADGroup = TargetResources[0].modifiedProperties[1].newValue
+  | where InvitedUser has_any ("CUSTOM DOMAIN NAME#", "#EXT#")
+  | mv-expand AADGroup = TargetResources[1].id to typeof(string)
+  | where AADGroup !in (GroupIDs)
+entityMappings:
+  - entityType: Account
+    fieldMappings:
+      - identifier: FullName
+        columnName: InvitedUser
+  - entityType: Account
+    fieldMappings:
+      - identifier: FullName
+        columnName: InitiatedByActionUserInformation
+  - entityType: IP
+    fieldMappings:
+      - identifier: Address
+        columnName: InitiatedByIPAdress
+version: 1.0.2
```

Comment:
Since this is a new template, the version should ideally be 1.0.0.

Reply:
Changed the version, to version 1.0.0